src/VirtoCommerce.CatalogModule.Data/Validation/CategoryPropertyNameValidator.cs

```diff
@@ -0,0 +1,46 @@
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using FluentValidation;
+using FluentValidation.Results;
+using VirtoCommerce.CatalogModule.Core.Model;
+using VirtoCommerce.CatalogModule.Core.Model.Search;
+using VirtoCommerce.CatalogModule.Core.Search;
+
+namespace VirtoCommerce.CatalogModule.Data.Validation
+{
+    public class CategoryPropertyNameValidator: AbstractValidator<CategoryPropertyValidationRequest>
+    {
+        private readonly IPropertySearchService _propertySearchService;
+
+        public CategoryPropertyNameValidator(IPropertySearchService propertySearchService)
+        {
+            _propertySearchService = propertySearchService;
+        }
+        public override async Task<ValidationResult> ValidateAsync(ValidationContext<CategoryPropertyValidationRequest> context, CancellationToken cancellation = default)
```

Comment:
Install Codemaid extension.

Reply:
Why we need to override ValidateAsync itself instead of using `MustAsync`? Like this: https://github.com/VirtoCommerce/vc-module-customer-export-import/blob/b42d24c002bd9b1a289520d1b58c92acfb24e7df/src/VirtoCommerce.CustomerExportImportModule.Data/Validation/ImportAccountValidator.cs#L93-L97

Reply:
Why we need to override ValidateAsync itself instead of using MustAsync? Like this: https://github.com/VirtoCommerce/vc-module-customer-export-import/blob/b42d24c002bd9b1a289520d1b58c92acfb24e7df/src/VirtoCommerce.CustomerExportImportModule.Data/Validation/ImportAccountValidator.cs#L93-L97